libevmasm/SemanticInformation.cpp

```diff
@@ -29,6 +29,128 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::evmasm;
 
+vector<SemanticInformation::Operation> SemanticInformation::readWriteOperations(Instruction _instruction)
+{
+	switch (_instruction)
+	{
+	case Instruction::SSTORE:
+	case Instruction::SLOAD:
+	case Instruction::MSTORE:
+	case Instruction::MSTORE8:
+	case Instruction::MLOAD:
+	case Instruction::REVERT:
+	case Instruction::RETURN:
+	case Instruction::EXTCODECOPY:
+	case Instruction::CODECOPY:
+	case Instruction::CALLDATACOPY:
+	case Instruction::RETURNDATACOPY:
+	case Instruction::KECCAK256:
+	case Instruction::LOG0:
+	case Instruction::LOG1:
+	case Instruction::LOG2:
+	case Instruction::LOG3:
+	case Instruction::LOG4:
+	{
+		Operation op;
+		if (memory(_instruction) == Effect::Write || storage(_instruction) == Effect::Write)
+			op.effect = Effect::Write;
+		else
+			op.effect = Effect::Read;
+
+		op.location =
+			(_instruction == Instruction::SSTORE || _instruction == Instruction::SLOAD) ?
+			Location::Storage :
+			Location::Memory;
+
+		if (_instruction == Instruction::EXTCODECOPY)
+			op.startParameter = 1;
+		else
+			op.startParameter = 0;
+
+		if (_instruction == Instruction::MSTORE || _instruction == Instruction::MLOAD)
+			op.lengthConstant = 32;
+		else if (_instruction == Instruction::MSTORE8)
+			op.lengthConstant = 1;
+		else if (
+			_instruction == Instruction::REVERT ||
+			_instruction == Instruction::RETURN ||
+			_instruction == Instruction::KECCAK256 ||
+			_instruction == Instruction::LOG0 ||
+			_instruction == Instruction::LOG1 ||
+			_instruction == Instruction::LOG2 ||
+			_instruction == Instruction::LOG3 ||
+			_instruction == Instruction::LOG4
+		)
+			op.lengthParameter = 1;
+		else if (_instruction == Instruction::EXTCODECOPY)
+			op.lengthParameter = 3;
+		else if (
+			_instruction == Instruction::CALLDATACOPY ||
+			_instruction == Instruction::CODECOPY ||
+			_instruction == Instruction::RETURNDATACOPY
+		)
+			op.lengthParameter = 2;
+		else if (_instruction == Instruction::SSTORE || _instruction == Instruction::SLOAD)
+			// Storage operations, length is unused / non-sensical
+			op.lengthConstant = 1;
```

Comment:
I don't think we should do this. Using length for storage feels error-prone.

Reply:
But it is actually correct. "Unknown" would be wrong IMO

Reply:
Actually in the case of `delegatecall`, storage length is set to unknown.